src/context/AuthContext.jsx

```diff
@@ -1,25 +1,99 @@
-import {createContext, useState} from "react";
+import {createContext, useEffect, useState} from "react";
+import {useNavigate} from "react-router-dom";
+import {jwtDecode} from "jwt-decode"
+import axios from "axios";
 
 export const AuthContext = createContext({});
 
-export function AuthContextProvider({children}){
-    const [isAuth, setIsAuth] = useState(0);
+export function AuthContextProvider({children}) {
+    const navigate = useNavigate()
+    useEffect(() => {
+        console.log("Context wordt gerefresht!")
+        if(localStorage.getItem("token")){
+            const decoded = jwtDecode(localStorage.getItem("token"))
+            const token = localStorage.getItem("token")
+            void fetchUserData(decoded, token)
+            console.log("er is een token", token)
+        }else{
+            setIsAuth({isAuth: false,
+                user:"",
+                status: "done",
+            })
+            console.log("er is geen token")
+        }
+    }, []);
 
-    function setAuthTrue(){
-        setIsAuth(true)
+    const [isAuth, setIsAuth] = useState({
+        isAuth: false, user: {
+            email: "",
+            password: "",
+            username: "",
+        },
+        status: "pending",
+    });
+
+
+    function logOut() {
+        setIsAuth({isAuth: false, user: ''});
+        localStorage.removeItem("token")
+        navigate('/')
+        console.log("Gebruiker is uitgelogd!")
+        setIsAuth({isAuth: false,
+            user:"",
+            status: "done",
+        })
```

Comment:
Je doet nu binnen één functie twee keer de `setIsAuth` functie aanroepen. Volgens mij is 1x voldoende
```suggestion
        localStorage.removeItem("token")
        navigate('/')
        console.log("Gebruiker is uitgelogd!")
        setIsAuth({
            isAuth: false,
            user:"",
            status: "done",
        })
```